homeassistant/components/bond/config_flow.py

```diff
@@ -0,0 +1,74 @@
+"""Config flow for Bond integration."""
+import logging
+
+from bond import Bond
+from requests.exceptions import ConnectionError as RequestConnectionError
+from simplejson import JSONDecodeError
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import CONF_ACCESS_TOKEN, CONF_HOST
+
+from .const import DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {vol.Required(CONF_HOST): str, vol.Required(CONF_ACCESS_TOKEN): str}
+)
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect."""
+
+    def authenticate(bond_hub: Bond) -> bool:
+        try:
+            bond_hub.getDeviceIds()
+            return True
+        except RequestConnectionError:
+            raise CannotConnect
+        except JSONDecodeError:
```

Comment:
Side note: It would be good if the library raised library specific exceptions. It's not easy for the library user to know to catch this kind of error.

Reply:
thanks, will try to make a change upstream